File: Data Structures & Algorithms/insert-new-interval/submission-1.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        ret = []
        n = len(intervals)
        start = 0

        i = 0
        while i < n and intervals[i][1] < newInterval[0]:
            ret.append(intervals[i])
            i+=1

        if len(ret) == n:
            ret.append(newInterval)
            return ret

        curr = newInterval        
        while i < n and curr[1] >= intervals[i][0]:
            curr[0] = min(curr[0],intervals[i][0])
            curr[1] = max(curr[1],intervals[i][1])
            i+=1
        ret.append(curr)
        while i < n:
            ret.append(intervals[i])
            i+=1
        return ret

File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int l = 0;
        int r = heights.length-1;
        int maxVol = 0;
        while(l < r){
            int h = Math.min(heights[l],heights[r]);
            int w = r-l;
            maxVol = Math.max(h*w, maxVol);
            if(heights[l]<heights[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return maxVol;
    }
}
